source/MaterialXGenShader/ShaderGraph.cpp

```diff
@@ -524,6 +530,9 @@ ShaderGraphPtr ShaderGraph::createSurfaceShader(
 
         // Connect graph input socket to the node input
         inputSocket->makeConnection(input);
+
+        // Share metadata.
+        inputSocket->setMetadata(input->getMetadata());
```

Comment:
Is it better to set and clear meta data when connections are made / broken? Otherwise isn't it possible to leave meta data after a connection is broken?

Reply:
Note that this only relates to metadata being exported to a shader during codegen. The shadergen graph is not a "live" data model that get's updated during user edits, as connections are made or broken, it's a single export from a fixed state of the user data model.

Reply:
And this metadata "transfer" is specifically for graph interface connections, for the same reason as above, so not applicable to connections inside the graph.